packages/client/subscriptions/TaskSubscription.ts

```diff
@@ -32,56 +32,40 @@ const subscription = graphql`
 
 const onNextHandlers = {
   UpdateTaskPayload: updateTaskTaskOnNext
-}
+} as const
+
+const updateHandlers = {
+  ChangeTaskTeamPayload: changeTaskTeamTaskUpdater,
+  CreateTaskPayload: createTaskTaskUpdater,
+  DeleteTaskPayload: deleteTaskTaskUpdater,
+  EditTaskPayload: editTaskTaskUpdater,
+  RemoveOrgUserPayload: removeOrgUserTaskUpdater,
+  UpdateTaskPayload: updateTaskTaskUpdater
+} as const
 
 const TaskSubscription = (
   atmosphere: Atmosphere,
   variables: TaskSubscriptionVariables,
   router: {history: RouterProps['history']}
 ) => {
-  const {viewerId} = atmosphere
   return requestSubscription<TTaskSubscription>(atmosphere, {
     subscription,
     variables,
     updater: (store) => {
       const payload = store.getRootField('taskSubscription') as any
       if (!payload) return
-      const type = payload.getValue('__typename')
+      const type = payload.getValue('__typename') as keyof typeof updateHandlers
       const context = {atmosphere, store: store as RecordSourceSelectorProxy<any>}
-      switch (type) {
-        case 'CreateTaskIntegrationPayload':
-          break
-        case 'ChangeTaskTeamPayload':
-          changeTaskTeamTaskUpdater(payload, context)
-          break
-        case 'CreateTaskPayload':
-          createTaskTaskUpdater(payload, context)
-          break
-        case 'DeleteTaskPayload':
-          deleteTaskTaskUpdater(payload, store)
-          break
-        case 'EditTaskPayload':
-          editTaskTaskUpdater(payload, store)
-          break
-        case 'RemoveOrgUserPayload':
-          removeOrgUserTaskUpdater(payload, store, viewerId)
-          break
-        case 'UpdateTaskPayload':
-          updateTaskTaskUpdater(payload, context)
-          break
-        case 'UpdateTaskDueDatePayload':
-          break
-        default:
-          console.error('TaskSubscription case fail', type)
-      }
+      const updater = updateHandlers[type]
+      updater?.(payload, context)
```

Comment:
I'd argue against it because it's not really an error if there's no handler, we expect some to have handlers & others only need to be handled by relay's internal logic. This one is just so so old that it doesn't operate like the newer ones (https://github.com/ParabolInc/parabol/blob/42ea3301ac0ec1ebd35a1323fd43bad8b0541d5a/packages/client/subscriptions/MeetingSubscription.ts#L103-L105)

Granted, it is a change to the runtime during a typing refactor, I probably should've let it be & fixed in another PR...